### ISSUE TITLE
Report inaccesible file on failed require

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14263]
Migrated from [rt.perl.org#123270](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123270) (status was 'resolved')

Searchable as RT123270$

**Comment by p5pRT** [comment]
### From @ppisar

As was appended to \[perl \#113422\]\, the error message due to an inaccessible
file while loading a module is confusing&#8203;:

&nbsp;    $ LANG=C perl \-I/root \-e 'require strict'
&nbsp;    Can't locate strict\.pm&#8203;:   Permission denied at \-e line 1\.

Attached patch improves the error message to be clear that /root/strict\.pm is
the problematic file\.

\-\- Petr

**Comment by p5pRT** [comment]
### From @ppisar

<details><summary>0001-Report-inaccesible-file-on-failed-require.patch</summary>

```diff
From 8817b5dd0a5e8f6fb7ad1c7826f6724cc83348dd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 21 Nov 2014 10:48:51 +0100
Subject: [PATCH] Report inaccesible file on failed require
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Commit 2433d39e6 (require should die if a file exists but can't be
read) made first failed opened file fatal as request in
[perl #113422]. However error message produced in that case is not
much helpful in identifying which file ound not been accessed:

    $ LANG=C perl -I/root -e 'require strict'
    Can't locate strict.pm:   Permission denied at -e line 1.

This patch adds the name of the failed file to the message to help
identify which @INC directory is erroneous:

    $ LANG=C ./perl -I/root -I./lib -e 'require strict'
    Can't locate strict.pm:   /root/strict.pm: Permission denied at -e line 1.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 pp_ctl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pp_ctl.c b/pp_ctl.c
index 59ad06e..4a20f34 100644
--- a/pp_ctl.c
+++ b/pp_ctl.c
@@ -4049,7 +4049,8 @@ PP(pp_require)
 	if (PL_op->op_type == OP_REQUIRE) {
 	    if(saved_errno == EMFILE || saved_errno == EACCES) {
 		/* diag_listed_as: Can't locate %s */
-		DIE(aTHX_ "Can't locate %s:   %s", name, Strerror(saved_errno));
+		DIE(aTHX_ "Can't locate %s:   %s: %s",
+		    name, tryname, Strerror(saved_errno));
 	    } else {
 	        if (namesv) {			/* did we lookup @INC? */
 		    AV * const ar = GvAVn(PL_incgv);
-- 
1.9.3


```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

On Fri Nov 21 02&#8203;:16&#8203;:38 2014\, ppisar wrote&#8203;:
> As was appended to \[perl \#113422\]\, the error message due to an
> inaccessible
> file while loading a module is confusing&#8203;:
> 
> $ LANG=C perl \-I/root \-e 'require strict'
> Can't locate strict\.pm&#8203;:   Permission denied at \-e line 1\.
> 
> Attached patch improves the error message to be clear that
> /root/strict\.pm is
> the problematic file\.
> 
> \-\- Petr

Thank you\.  Applied as e2ce095\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'pending release'

**Comment by p5pRT** [comment]
### From @khwilliamson

Thanks for submitting this ticket

The issue should be resolved with the release today of Perl v5\.22\, available at http&#8203;://www\.perl\.org/get\.html
If you find that the problem persists\, feel free to reopen this ticket

\-\- 
Karl Williamson for the Perl 5 porters team

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'pending release' to 'resolved'
